interface.rb

```diff
@@ -0,0 +1,31 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError
+  end
+end
+
+class BaseDecorator
```

Comment:
- [ ] The `BaseDecorator class` inherits from the `Nameable class`, kindly fix that. Please make sure you also set the getter and setter methods for the nameable object. Something like so: 
 `class BaseDecorator < Nameable` , `attr_accessor :nameable`.

Reply:
Thank you @Mihndim2020 for the tips.